storage/engine/rocksdb/db.cc

```diff
@@ -39,33 +39,84 @@
 extern "C" {
 #include "_cgo_export.h"
 
-struct DBBatch {
+struct DBEngine {
+  rocksdb::DB* const rep;
+
+  DBEngine(rocksdb::DB* r)
+      : rep(r) {
+  }
+  virtual ~DBEngine() { }
+
+  virtual DBStatus Put(DBSlice key, DBSlice value);
```

Comment:
It looks like these methods can be pure in the base class and the implementations moved to `DBImpl` (in which case disregard my other comment about the name of `DBImpl`)